src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state }; // Create a clone of the initial state
+
+  for (const action of actions) {
+    const { type } = action;
+
+    if (type === 'addProperties') {
+      const { extraData } = action;
+
+      currentState = {
```

Comment:
Consider using Object.assign instead of spread syntax to merge objects. Example: currentState = Object.assign(currentState, extraData);